Modules/Microsoft365DSC/DSCResources/MSFT_IntuneDeviceCleanupRule/MSFT_IntuneDeviceCleanupRule.psm1

```diff
@@ -0,0 +1,381 @@
+function Get-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Collections.Hashtable])]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [System.Boolean]
+        $Enabled,
+
+        [Parameter()]
+        [System.Int32]
+        [ValidateRange(30, 270)]
+        $DeviceInactivityBeforeRetirementInDays,
+
+        [Parameter()]
+        [System.String]
+        [ValidateSet('Absent', 'Present')]
+        $Ensure = 'Present',
+
+        [Parameter()]
+        [System.Management.Automation.PSCredential]
+        $Credential,
+
+        [Parameter()]
+        [System.String]
+        $ApplicationId,
+
+        [Parameter()]
+        [System.String]
+        $TenantId,
+
+        [Parameter()]
+        [System.Management.Automation.PSCredential]
+        $ApplicationSecret,
+
+        [Parameter()]
+        [System.String]
+        $CertificateThumbprint,
+
+        [Parameter()]
+        [Switch]
+        $ManagedIdentity
+    )
+
+    if ($Enabled -and $DeviceInactivityBeforeRetirementInDays -eq 0)
+    {
+        throw [System.ArgumentException]::new('DeviceInactivityBeforeRetirementInDays must be greater than 30 and less than 270 when Enabled is set to true.')
+    }
+
+    Write-Verbose -Message "Checking for the Intune Device Cleanup Rule"
+    $ConnectionMode = New-M365DSCConnection -Workload 'MicrosoftGraph' `
+        -InboundParameters $PSBoundParameters
+
+    #Ensure the proper dependencies are installed in the current environment.
+    Confirm-M365DSCDependencies
+
+    #region Telemetry
+    $ResourceName = $MyInvocation.MyCommand.ModuleName -replace 'MSFT_', ''
+    $CommandName = $MyInvocation.MyCommand
+    $data = Format-M365DSCTelemetryParameters -ResourceName $ResourceName `
+        -CommandName $CommandName `
+        -Parameters $PSBoundParameters
+    Add-M365DSCTelemetryEvent -Data $data
+    #endregion
+
+    $nullResult = $PSBoundParameters
+    $nullResult.Ensure = 'Absent'
+
+    try
+    {
+        $url = "https://graph.microsoft.com/beta/deviceManagement/managedDeviceCleanupSettings"
```

Comment:
Let's not hardcode URLs and instead use the $Global:MSCloudLoginProfile.MicrosoftGraph values to generalize this. Otherwise the code doesn't work for sovereign clouds. Also, we should use the cmdlets instead of direct graph calls wherever possible. In this case, we should call (Get-MgBetaDeviceManagement).ManagedDeviceCleanupSettings

Reply:
As suggested I tried the `Get-MgBetaDeviceManagement` cmdlet, but that one does not return a value for the `DeviceInactivityBeforeRetirementInDays` property. It simply returns `$null`. So unfortunately, we can't use that one yet. Might raise an issue for that. 
![image](https://github.com/microsoft/Microsoft365DSC/assets/71251572/1d6d7a9b-1fca-44a4-b448-3e2cf995d2f2)

Reply:
I will open an issue a bit later, stay tight. In the meantime, I updated the code to use the `$Global:MSCloudLoginConnectionProfile.MicrosoftGraph.ResourceUrl` property. Tested and verified, works as intended.